src/libslic3r/LayerRegion.cpp

```diff
@@ -113,28 +113,24 @@ void LayerRegion::make_perimeters(
         auto perimeters_begin      = uint32_t(m_perimeters.size());
         auto gap_fills_begin       = uint32_t(m_thin_fills.size());
         auto fill_expolygons_begin = uint32_t(fill_expolygons.size());
-        if (this->layer()->object()->config().perimeter_generator.value == PerimeterGeneratorType::Arachne && !spiral_vase)
-            PerimeterGenerator::process_arachne(
+        bool use_arachne = this->layer()->object()->config().perimeter_generator.value == PerimeterGeneratorType::Arachne && !spiral_vase;
+        bool dithered_layer    = this->layer()->dithered;
+        // arachne width averaging is not so good for thin areas of dithered layers becuase it results in routh attachment/seams of 
+        // dithered layers to non-dithered (regular) layers. On the other hand for very narrow dithered layers classic algothim  
+        // sometimes makes no perimeters at all and in this case we better employ arachne algorithm (even if user selected classic).
+        if (!use_arachne || dithered_layer)
```

Comment:
It's better not to switch Arachne/Classic calls order, neither reformat parameter list, as it causes problem with further merges.

Reply:
While changing the formatting of parameters could be done and is not a big deal, the alternative to switching Arachne/Classic calls order was considered and turned out to be much worse.

Reply:
Even restore format will be good, as I had problem in merging another PR over dithering. But what is problem with call order? Aren't two calls mutually exclusive? I guess this

```
if (dithered_layer && m_perimeters.empty() || !dithered_layer && use_arachne)
       PerimeterGenerator::process_arachne(
.........
else
       PerimeterGenerator::process_classic(
```
should work, or do I miss something?

Reply:
For dithered layers, Classic paths connect to adjacent non-dithers layers much better than Arachne.  My code for dithered layers attempted to use Classic first, then if Classic failed, use Arachne.  Observe m_perimeters.empty() as a check for Classic failure and two if statements as opposed to if/else.